nifi-commons/nifi-record-path/src/main/java/org/apache/nifi/record/path/functions/Base64Encode.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.record.path.functions;
+
+import org.apache.nifi.record.path.FieldValue;
+import org.apache.nifi.record.path.RecordPathEvaluationContext;
+import org.apache.nifi.record.path.StandardFieldValue;
+import org.apache.nifi.record.path.paths.RecordPathSegment;
+
+import java.io.UnsupportedEncodingException;
+import java.util.Base64;
+import java.util.stream.Stream;
+
+public class Base64Encode extends RecordPathSegment {
+    private final RecordPathSegment recordPath;
+
+    public Base64Encode(final RecordPathSegment recordPath, final boolean absolute) {
+        super("base64Encode", null, absolute);
+        this.recordPath = recordPath;
+    }
+
+    @Override
+    public Stream<FieldValue> evaluate(final RecordPathEvaluationContext context) {
+        final Stream<FieldValue> fieldValues = recordPath.evaluate(context);
+        return fieldValues.filter(fv -> fv.getValue() != null)
+                .map(fv -> {
+
+                    Object value = fv.getValue();
+                    if (value instanceof String) {
+                        try {
+                            return new StandardFieldValue(Base64.getEncoder().encodeToString(value.toString().getBytes("UTF-8")), fv.getField(), fv.getParent().orElse(null));
+                        } catch (final UnsupportedEncodingException e) {
+                            return null;    // won't happen.
+                        }
+                    } else if (value instanceof byte[]) {
```

Comment:
@markap14 Are there other representations of byte arrays I have to deal with here? Byte[], ByteBuffer, etc.? I thought by the time it's a FieldValue it would be a byte[] but I might be mistaken...

Reply:
@mattyb149  Is there anyone else other than @markap14 that can provide the answer here? If this patch is appropriate, it is unfortunate to see it become stale.

Reply:
I don't think there is a problem starting with string and byte[]

Reply:
Sorry @mattyb149 I never saw your comment. I think String and byte[] are certainly the most important. You may also want to consider Character. I'll leave it up to you whether you think it makes sense to allow for arrays (i.e., allow base64Encode(/names) where /names is an Array of type String, aka String[])....

Reply:
I tend to agree with @ottobackwards. @mattyb149 @markap14 I'll merge later today if neither of you wants me to hold off because the PR LGTM.